src/abstractdataframe/abstractdataframe.jl

```diff
@@ -189,7 +189,13 @@ rename(df, :i => :A, :x => :X)
 rename(df, [:i => :A, :x => :X])
 rename(df, Dict(:i => :A, :x => :X))
 rename(x -> Symbol(uppercase(string(x))), df)
+rename(df) do x
+    x |> string |> uppercase |> Symbol
+end
 rename!(df, Dict(:i =>: A, :x => :X))
+rename!(df) do x
+    x |> string |> uppercase |> Symbol
```

Comment:
here it maybe better to use `lowercase` 😄 as otherwise this is a no-op as we used `rename!` in the line above?